File: Cpp/2956_find_common_elements_between_two_arrays.cpp
// https://leetcode.cn/problems/find-common-elements-between-two-arrays/
        
class Solution {
public:
    vector<int> findIntersectionValues(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> s1(nums1.begin(), nums1.end()), s2(nums2.begin(), nums2.end());
        int cnt1 = 0, cnt2 = 0;
        for (int num: nums1) {
            if (s2.find(num) != s2.end()) cnt1 += 1;
        }
        for (int num: nums2) {
            if (s1.find(num) != s1.end()) cnt2 += 1;
        }
        return vector<int> {cnt1, cnt2};
    }
};

File: Python3/2956_find_common_elements_between_two_arrays.py
# https://leetcode.cn/problems/find-common-elements-between-two-arrays/

class Solution:
    def findIntersectionValues(self, nums1, nums2):
        """
        Inputs:
            nums1, nums2: list[int]
        Outputs:
            res: list[int]
        """
        s1, s2 = set(nums1), set(nums2)
        cnt1, cnt2 = 0, 0
        for num in nums1:
            if num in s2:
                cnt1 += 1
        for num in nums2:
            if num in s1:
                cnt2 += 1
        return [cnt1, cnt2]